commit 3d3d87f3a70089f645facb2bff5a0b93ec0a0efd
Author: Marcin Slusarz <marcin.slusarz@gmail.com>
Date:   Tue Jan 12 15:38:29 2010 +0100

    nouveau: disable flush_notify on channel_free
    
    We don't want do call flush_notify when we will FIRE the RING
    a couple of lines later, because grobj bound to this channel
    might be already freed.

diff --git a/nouveau/nouveau_channel.c b/nouveau/nouveau_channel.c
index 14cf4c6..01d6681 100644
--- a/nouveau/nouveau_channel.c
+++ b/nouveau/nouveau_channel.c
@@ -107,6 +107,7 @@ nouveau_channel_free(struct nouveau_channel **chan)
 	if (!chan || !*chan)
 		return;
 	nvchan = nouveau_channel(*chan);
+	(*chan)->flush_notify = NULL;
 	*chan = NULL;
 	nvdev = nouveau_device(nvchan->base.device);
 
